File: data_struct/stack.cpp
#include <iostream>
#include "stack.h"

using namespace std;
Stack::Stack(int stkSize){
	maxSize=stkSize;
}

bool Stack::isFull(){
	bool full = (Stk.listLen() == maxSize);
	return full;
}

bool Stack::isEmpty(){
	bool empty = (Stk.listLen() == 0);
	return empty;
}

void Stack::Push(int data){
	if (!(Stack::isFull())){
		Stk.insNode(data);
		Stk.printList();
	}
	else
		cout<<"Stack is full"<<endl;
}

void Stack::Pop(){
	if (!(Stack::isEmpty())){
		Stk.delNode();
		Stk.printList();
	}
	else
		cout<<"Stack is Empty"<<endl;
}
	
	
	

File: pyads/stack.py
from list import SingleLL

class Stack:
	def __init__(self, maxSize=25):
		self.maxSize=maxSize
		self.stk = SingleLL()
		self.stkSize = 0
	
	def push(self, data):
		if not self.isFull():
			self.stk.append(data)
			self.stkSize += 1
		else:
			print("Stack is Full")
	
	def pop(self):
		if not self.isEmpty():
			value = self.stk.head.value
			self.stk.delFirstNode()
			self.stkSize -= 1
			return value
		else:
			print("Stack is Empty")
	
	def printStack(self):
		self.stk.printList()

	def isFull(self):
		return (self.stkSize == self.maxSize)
	
	def isEmpty(self):
		return (self.stkSize == 0)